brian2/equations/equations.py

```diff
@@ -433,6 +437,25 @@ def __init__(self, type, varname, dimensions, var_type=FLOAT, expr=None,
                                                       if variable =='xi' or variable.startswith('xi_')]),
                                     doc='Stochastic variables in the RHS of this equation')
 
+
+    _state_tuple = property(lambda self: (self.type, self.varname,
+                                          self.dim, self.var_type,
+                                          self.expr, tuple(self.flags)),
+                            doc='A tuple representing the full state of this '
+                                'object, used for hashing and equality '
+                                'testing.')
```

Comment:
As mentioned above, I don't think we can automate this in a straightforward way (we could have a function where we define which attributes should be considered for caching and which not, but then I don't think this would make things clearer).